eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/eventlist/EventListActivity.java

```diff
@@ -80,20 +79,20 @@ protected void onCreate(Bundle savedInstanceState)
         mScreenView = new EvenListScreenViewImpl(this);
         mScreenView.init();
 
-        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
-
         mApiService = new FutureLocalServiceConnection<>(this,
                 new Intent("com.schedjoules.API").setPackage(getPackageName()));
 
-        mEventListItems = new EventListItemsImpl();
-        mListItemsProvider = retainedObjects.getOr(0,
-                new EventListItemsProviderImpl(mApiService, mEventListItems,
-                        new EventListBackgroundMessage(this), new EventListLoadingIndicatorOverlay(this)));
+        RetainedObjects retainedObjects = new Restoring(getLastCustomNonConfigurationInstance());
+        mListItemsProvider = retainedObjects.getOr(0, new EventListItemsProviderImpl(new EventListItemsImpl()));
+
+        mListItemsProvider.setApiService(mApiService);
```

Comment:
I can see why you need these setters now, but I don't like the solution. In general I'm curious if we can't just create a `Parcelable` state that's retained in a Bundle and used to create a new instance when the activity is resumed.
I'm doing something similar in the oauth2-essentials [`AuthorizationCodeGrant`](https://github.com/dmfs/oauth2-essentials/blob/master/src/main/java/org/dmfs/oauth2/client/grants/AuthorizationCodeGrant.java)

In some cases it might be required to pass the `AuthorizationCodeGrant` to another component using some sort of serialization (or even just to retain it during configuration changes in an activity). That's a problem because it needs to know the complex `OAuth2Client` object (which might need a context to be created). To overcome this it has a method `state()` that returns a `Serializable` factory object. After de-serialization this factory can be used to recreate the original `AuthorizationCodeGrant`. Essentially it works like this
```
AuthorizationCodeGrant authCodeGrant = new AuthorizationCodeGrant(client, scope, randomstateparam);
// do something with authCodeGrant

// need to create a serializable state
OAuth2InteractiveGrant.OAuth2GrantState state = authCodeGrant.state();
// serialize state

// after de-serializing "state" you can use it to recreate the AuthorizationCodeGrant
AuthorizationCodeGrant sameAuthCodeGrant = state.grant(client);
```
That's certainly a cleaner way.

The current method using `Restoring` is error prone, not only because you can easily leak the context. Also, using integers to refer to the retained data is easy to get wrong.

That being said, I see that we need to retain the `ResultPage` or the `ApiQuery` for the next page using the custom non configuration parameters, because these can not be serialized or parcelled (yet). But we should retain as few (and small) objects as possible that way.